lib/Core/Inference/SymbolContextResolver.php

```diff
@@ -457,8 +457,7 @@ private function resolveParameterFromReflection(Frame $frame, MethodDeclaration
 
     private function resolveNumericLiteral(NumericLiteral $node): SymbolContext
     {
-        // note hack to cast to either an int or a float
-        $value = $node->getText() + 0;
+        $value = eval(sprintf('return %s;', $node->getText()));
```

Comment:
hm, i wonder if there is a way without using eval

Reply:
e.g. check if the string has non-numeric chars and then try and convert it from octal

Reply:
Well, of course we can have 4 `if`, if you prefer. Do you?

Reply:
I've updated the code to be without `eval`

Reply:
probably better tbh - the chance of something bad happening with `eval` here are very very remote, but possible all the same

maybe just regex?

```
decimal     : [1-9][0-9]*(_[0-9]+)*
            | 0

hexadecimal : 0[xX][0-9a-fA-F]+(_[0-9a-fA-F]+)*

octal       : 0[0-7]+(_[0-7]+)*

binary      : 0[bB][01]+(_[01]+)*

integer     : decimal
            | hexadecimal
            | octal
            | binary
```

(https://www.php.net/manual/en/language.types.integer.php)